io/net/virt-net/network_virtualization.py

```diff
@@ -456,11 +456,11 @@ def test_rmdev_viosfailover(self):
         if vnic_backing_device:
             self.validate_vios_command('rmdev -l %s' % vnic_backing_device, 'Defined')
 
-        after = self.get_active_device_logport(self.slot_num[0])
-        self.log.debug("Active backing device after: %s", after)
-
-        if before == after:
-            self.fail("failover not occur")
+        for backing_dev in self.backing_dev_list().splitlines():
+            if backing_dev.startswith('%s,' % self.slot_num[0]):
+                backing_dev = backing_dev.strip('%s,"' % self.slot_num[0])
+                if not 'Powered Off' in backing_dev:
+                    self.fail("failover not occur")
```

Comment:
nit pick, message can be changed to "Failover did not occur" or "Failover failed".